e2e: sporadic OCI kill test failures (EL7/8 ARM64)

**Version of Singularity**

3.11.0

**Describe the bug**

`kill` subtest, and subsquent tests can fail...


```
        --- FAIL: TestE2E/SEQ (83.10s)
            --- FAIL: TestE2E/SEQ/OCI (4.65s)
                --- FAIL: TestE2E/SEQ/OCI/ordered (4.65s)
                    --- FAIL: TestE2E/SEQ/OCI/ordered/systemd (2.32s)
                        --- FAIL: TestE2E/SEQ/OCI/ordered/systemd/basic (0.96s)
                            --- FAIL: TestE2E/SEQ/OCI/ordered/systemd/basic/kill (0.04s)
                            --- FAIL: TestE2E/SEQ/OCI/ordered/systemd/basic/delete (0.02s)
                            --- FAIL: TestE2E/SEQ/OCI/ordered/systemd/basic/state_fail (0.02s)
                            --- FAIL: TestE2E/SEQ/OCI/ordered/systemd/basic/kill_fail (0.02s)
                    --- FAIL: TestE2E/SEQ/OCI/ordered/cgroupfs (2.28s)
                        --- FAIL: TestE2E/SEQ/OCI/ordered/cgroupfs/basic (0.92s)
                            --- FAIL: TestE2E/SEQ/OCI/ordered/cgroupfs/basic/kill (0.04s)
                            --- FAIL: TestE2E/SEQ/OCI/ordered/cgroupfs/basic/delete (0.02s)
                            --- FAIL: TestE2E/SEQ/OCI/ordered/cgroupfs/basic/state_fail (0.02s)
                            --- FAIL: TestE2E/SEQ/OCI/ordered/cgroupfs/basic/kill_fail (0.02s)
    FAIL
```

The OCI container is still seen running:

```
    === RUN   TestE2E/SEQ/OCI/ordered/systemd/basic/kill
        oci.go:293: Running command "/usr/bin/singularity oci kill cb0baddc-092d-4b69-b963-4b1c0747fe93 KILL"
        singularitycmd.go:698: Running command "/usr/bin/singularity oci state cb0baddc-092d-4b69-b963-4b1c0747fe93"
        oci.go:45: bad container state returned, got running instead of stopped
```

This is happening frequently on CentOS 7 arm64, RHEL8 arm64. it is not really observed... or may just be very infrequent, elsewhere.

--- a/e2e/oci/oci.go
+++ b/e2e/oci/oci.go
@@ -9,6 +9,7 @@ import (
 	"encoding/json"
 	"os"
 	"testing"
+	"time"
 
 	"github.com/google/uuid"
 	"github.com/opencontainers/runtime-spec/specs-go"
@@ -34,6 +35,8 @@ type ctx struct {
 }
 
 func (c *ctx) checkOciState(t *testing.T, containerID, state string) {
+	ok := false
+
 	checkStateFn := func(t *testing.T, r *e2e.SingularityCmdResult) {
 		s := &ociruntime.State{}
 		if err := json.Unmarshal(r.Stdout, s); err != nil {
@@ -42,8 +45,10 @@ func (c *ctx) checkOciState(t *testing.T, containerID, state string) {
 			return
 		}
 		if s.Status != specs.ContainerState(state) {
-			t.Errorf("bad container state returned, got %s instead of %s", s.Status, state)
+			t.Logf("bad container state returned, got %s instead of %s", s.Status, state)
+			return
 		}
+		ok = true
 	}
 
 	c.env.RunSingularity(
@@ -53,6 +58,24 @@ func (c *ctx) checkOciState(t *testing.T, containerID, state string) {
 		e2e.WithArgs(containerID),
 		e2e.ExpectExit(0, checkStateFn),
 	)
+	if ok {
+		return
+	}
+
+	// State transition is slow on EL7/8 ARM64, check again in 1s
+	// https://github.com/sylabs/singularity/issues/1437
+	t.Logf("Retrying state check in 1s...")
+	time.Sleep(time.Second)
+	c.env.RunSingularity(
+		t,
+		e2e.WithProfile(e2e.RootProfile),
+		e2e.WithCommand("oci state"),
+		e2e.WithArgs(containerID),
+		e2e.ExpectExit(0, checkStateFn),
+	)
+	if !ok {
+		t.Errorf("Expected state %s not found after retrying", state)
+	}
 }
 
 func genericOciMount(t *testing.T, c *ctx) (string, func()) {

